docs/src/introduction/overview.md

```diff
@@ -22,107 +22,143 @@ There are four types of Sway programs:
+For simplicity, we will define the ABI natively in the contract.
 
-In the first line, we declare the name of this ABI: `Wallet`. To import this ABI into either a script or another contract for calling the contract, or the contract to implement the ABI, you would use `use wallet_lib::Wallet;`.
+```sway
+abi Counter {
+    #[storage(read, write)]
+    fn increment();
 
-In the second line we declare an ABI method called `receive_funds` which, when called, should receive funds into this wallet. This method takes no parameters and does not return anything.
+    #[storage(read)]
+    fn counter() -> u64;
+}
+```
 
-> **Note**: We are simply defining an interface here, so there is no function body or implementation of the function. We only need to define the interface itself. In this way, ABI declarations are similar to [trait declarations](../advanced/traits.md).
+### Going line by line
 
-In the third line we declare another ABI method, this time called `send_funds`. It takes two parameters: the amount to send, and the address to send the funds to.
+`#[storage(read, write)]` is an annotation which denotes that this function has the permissions to read and write a value in storage.
 
-### Implementing the ABI Methods in `wallet_contract`
+`fn increment()` - We're introducing the functionality to increment and denoting it shouldn't return any value.
 
-Now that we've defined the interface, let's discuss how to use it. We will start by implementing the above ABI for a specific contract.
+`#[storage(read)]` is an annotation which denotes that this function has the permissions to read values in storage.
 
-To do this, navigate to the `wallet_contract` directory that you created with `forc` previously.
+`fn counter() -> u64;` - We're introducing the functionality to to increment the counter and denoting the function's return value.
 
-First, you need to import the `Wallet` declaration from the last step. Open up `Forc.toml`. It should look something like this:
+### Implement ABI
 
-```toml
-[project]
-authors = ["user"]
-entry = "main.sw"
-license = "Apache-2.0"
-name = "wallet_contract"
+Below your ABI definition, you will write the implementation of the functions defined in your ABI.
 
-[dependencies]
+```sway
+impl Counter for Contract {
+    #[storage(read)]
+    fn counter() -> u64 {
+    return storage.counter;
+  }
+    #[storage(read, write)]
+    fn increment(){
+        storage.counter = storage.counter + 1;
+    }
+}
 ```
 
-Include the `wallet_lib` project as a dependency by adding the following line to the bottom of the file:
+> Note: `return storage.counter;` is equivalent to `storage.counter`.
 
-```toml
-wallet_lib = { path = "../wallet_lib" }
-```
+### What we just did
 
-Now, open up `main.sw` in `wallet_contract/src` and copy and paste the following code:
+`#[storage(read)]` is an annotation which denotes that this function has the permissions to read values in storage.
```

Comment:
do we need to repeat these lines each time?

Reply:
You right, I wasn't sure but I will remove the second instance of these